### PR TITLE
ensure requests are good

### PR DESCRIPTION
**Pull request by BRShapiro**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

make sure only users can request their own profile info

### CODE REVIEW

File: src/domain/profile/profile.validator.ts [hunk 1]
```diff
@@ -5,9 +5,25 @@ export default class ProfileValidator {
...
+    return {
+      errors: errors,
+      isValid: (errors.length == 0)
+    }
+  }
+
   validate(profile) {
     let errors = []
     this.validateName(profile.firstName, profile.lastName, errors)
+    // this.validateInspiration(profile.inspiration, errors)
```
Comment (markd315) [inline src/domain/profile/profile.validator.ts]: Shouldn't we turn these on before merging?

File: src/domain/profile/profile.validator.ts [hunk 2]
```diff
@@ -27,4 +43,26 @@ export default class ProfileValidator {
       errors.push("Last name must be between 4 and 20 characters in length.")
     }
   }
+
+  private validateInspiration(inspiration, errors) {
```
Comment (markd315) [inline src/domain/profile/profile.validator.ts]: This does just strike me as a field that should be optional, but if you have specific requirements from someone else then feel free to ignore me.

File: src/domain/profile/profile.validator.ts [hunk 3]
```diff
@@ -27,4 +43,26 @@ export default class ProfileValidator {
...
+    if (inspiration == undefined) {
+      errors.push("Inspiration is a required field. Must not be null")
+    } else if (inspiration.length == 0) {
+      errors.push("Inspiration is required and may not be left blank.")
+    }
+  }
+
+  private validateGradYear(gradYear, errors) {
+    if (<number> gradYear < 2018) {
```
Comment (markd315) [inline src/domain/profile/profile.validator.ts]: Do not hardcode this, take a look at the following sample code and probably change the date format mainly

```
private validateHangoutDate(date, errors) {
        let dateFormats = ["YYYY-MM-DD HH:mm", "YYYY-MM-DD"]; //prefer 24-hour time
        if (date == null || date == undefined || date == "") {
            return //Allow blank dates
        }
        let moment = require('moment');
        let parsedDate = moment(date, dateFormats);
        if (!parsedDate.isValid()) {
            errors.push("Hangout date is invalid.")
        } else if (parsedDate.isBefore()) { //If a past date is given
            errors.push("Hangout date must be in the future.")
        }
    }
```

File: src/domain/profile/profile.validator.ts (hunk 2 above)
Comment (audgster) [reply src/domain/profile/profile.validator.ts]: It's not really optional

**Comment by BRShapiro** [comment]
I realize that my branch was behind. I didn't mean to add the validation for school and grad year.

The point of this PR is to update how we store UIDs and to validate that requests to get and delete profiles is made from a validated source, everything else I removed.

File: src/domain/profile/profile.validator.ts (hunk 1 above)
Comment (markd315) [reply src/domain/profile/profile.validator.ts]: These are still in the diff

**Pull request closed without merging**